hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/service/Bound.java

```diff
@@ -103,7 +143,17 @@ public boolean hasLowerAndUpper() {
     }
 
     public boolean hasEqualBounds() {
-        return hasLowerAndUpper() && adjustLowerBound() == adjustUpperBound();
+        return hasLowerAndUpper() && getAdjustedLowerRangeValue() == adjustUpperBound();
+    }
+
+    // Returns a new bound with only a lower rangeParameter
+    public Bound toLower(Bound bound) {
+        return createBound(bound.getLower());
+    }
+
+    // Returns a new bound with only an upper rangeParameter
+    public Bound toUpper(Bound bound) {
+        return createBound(bound.getUpper());
```

Comment:
This was not what I suggested. It doesn't make sense to pass in the same bound that you're invoking it on.

```suggestion
    public Bound toLower() {
        return createBound(getLower());
    }

    // Returns a new bound with only an upper rangeParameter
    public Bound toUpper() {
        return createBound(getUpper());
```

Reply:
Doh, thanks, much better. I have fixed it now.